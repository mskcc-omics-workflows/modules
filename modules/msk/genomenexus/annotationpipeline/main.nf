process GENOMENEXUS_ANNOTATIONPIPELINE {
    tag "$meta.id"
    label 'process_high'


    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'ghcr.io/msk-access/genomenexus_annotation-pipeline:1.0.4':
        'ghcr.io/msk-access/genomenexus_annotation-pipeline:1.0.4' }"

    input:
    tuple val(meta), path(input_maf)

    output:
    tuple val(meta), path("${meta.control_id}_${meta.case_id}_annotated.maf"), emit: annotated_maf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    java -jar /genome-nexus-annotation-pipeline/annotationPipeline/target/annotationPipeline.jar --filename ${input_maf} --output-filename ${meta.id}_annotated.maf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        genomenexus: 'annotationpipeline'
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${meta.control_id}_${meta.case_id}_annotated.maf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        genomenexus: 'annotationpipeline'
    END_VERSIONS
    """
}
