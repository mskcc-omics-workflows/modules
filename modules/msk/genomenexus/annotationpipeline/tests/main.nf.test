
// nf-core modules test genomenexus/annotationpipeline
nextflow_process {

    name "Test Process GENOMENEXUS_ANNOTATIONPIPELINE"
    script "../main.nf"
    process "GENOMENEXUS_ANNOTATIONPIPELINE"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "genomenexus"
    tag "genomenexus/annotationpipeline"


    test("test normal ") {


        when {
            process {
                """

                input[0] = [
                    [ id:'test'], // meta map
                    file(params.test_data_mskcc['genome_nexus']['test_maf'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }


    test("test normal - stub") {

        options "-stub"

        when {
            process {
                """


                input[0] = [
                    [ id:'test'], // meta map
                    file(params.test_data_mskcc['genome_nexus']['test_maf'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success }

            )
        }

    }

}
