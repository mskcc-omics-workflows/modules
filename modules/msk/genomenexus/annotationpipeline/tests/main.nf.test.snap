{
    "test normal ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_annotated.maf:md5,4547e768939e82f6d1f44e1c64b64c3a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d11163e0a6bf41b0a3de47036e5f90e3"
                ],
                "annotated_maf": [
                    [
                        {
                            "id": "test"
                        },
                        "test_annotated.maf:md5,4547e768939e82f6d1f44e1c64b64c3a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d11163e0a6bf41b0a3de47036e5f90e3"
                ]
            }
        ],
        "timestamp": "2024-06-24T13:11:46.963057"
    }
}